packages/fluentui/react-bindings/src/styles/resolveVariables.ts

```diff
@@ -37,9 +37,18 @@ const resolveVariables = (
     const handlingDisplayName = effectiveDisplayNames[effectiveDisplayNames.length - 1];
 
     if (!variablesThemeCache[handlingDisplayName]) {
-      variablesThemeCache[handlingDisplayName] = mergeComponentVariables(
-        ...effectiveDisplayNames.map(displayName => theme.componentVariables[displayName]),
-      )(theme.siteVariables);
+      // A short circle to avoid additional merging for non-composed components
+      if (effectiveDisplayNames.length === 1) {
+        variablesThemeCache[handlingDisplayName] =
+          typeof theme.componentVariables[handlingDisplayName] === 'function'
```

Comment:
`callable(theme.componentVariables[handlingDisplayName])(theme.siteVariables)`?

Reply:
Yup, it will create an additional frame but it's not important